scheduler/signals.py

```diff
@@ -118,20 +118,31 @@ def notify_users_shift_change(sender, instance, **kwargs):
 @receiver(post_save, sender=ShiftMessageToHelpers)
 def send_shift_message_to_helpers(sender, instance, **kwargs):
     for recipient in instance.recipients.all():
-        message = render_to_string(
-            "shift_message_to_helpers.html",
-            dict(message=instance.message, recipient=recipient, shift=instance.shift),
-        )
-        subject = _(
-            "Volunteer-Planner: A Message from shift manager of {shift_title}".format(
-                shift_title=instance.shift.task.name.strip
-            )
-        )
-        mail = EmailMessage(
-            subject=subject,
-            body=message,
-            to=[recipient.user.email],
-            from_email="noreply@volunteer-planner.org",
-            reply_to=instance.sender.user.email,
-        )
-        mail.send()
+        if instance.sender.user.email:
+            try:
+                message = render_to_string(
+                    "shift_message_to_helpers.html",
+                    dict(
+                        message=instance.message,
+                        recipient=recipient,
+                        shift=instance.shift,
+                        sender_email=instance.sender.user.email,
+                    ),
+                )
+                subject = _(
+                    "Volunteer-Planner: A Message from shift "
+                    "manager of {shift_title}".format(
+                        shift_title=instance.shift.task.name
+                    )
+                )
+                if message:
+                    mail = EmailMessage(
+                        subject=subject,
+                        body=message,
+                        to=[recipient.user.email],
+                        from_email="noreply@volunteer-planner.org",
+                        reply_to=(instance.sender.user.email,),
+                    )
+                    mail.send()
+            except Exception as e:
+                logger.info("send_shift_message_to_helpers: message not successful", e)
```

Comment:
You mean it's better to raise an Error? What I want to prevent is, that the loop will break. It should try to send the other emails at least

Reply:
No, I mean: log a more severe message, ie. `logger.error(...)` because the log level is set to `ERROR` in `voulunteer_planner.settings.production`. This means, anything below `ERROR` will be surpressed in production and operations might never learn, when there is a problem at all.

Reply:
in other words: this logging only works on your machine ;-)